<launch>
  <arg name="wheel_name"    default="wheel"/>
  <arg name="control_mode"  default="velocity"/>
  <arg name="sample_rate"   default="12"/>
  <arg name="swerve_id"     default="1"/>
  <arg name="wheel_id"      default="2"/>
  <arg name="wheel_reverse" default="false"/>
  <arg name="sim"           default="false"/>
  
  <!-- <param name="robot_description" textfile="$(find swerve_drive_description)/urdf/single_wheel.xacro" /> -->
  <arg name="urdf_path" default="$(find swerve_drive_description)/urdf/single_wheel.xacro"/>
  <param name="robot_description" 
    command="$(find xacro)/xacro '$(arg urdf_path)'" />
  
  <node name="wheel_controller" pkg="wheel_controller" type="wheel_controller_node" output="screen">
    <param name="wheel_name"    value="$(arg wheel_name)"/>
    <param name="control_mode"  value="$(arg control_mode)"/>
    <param name="sample_rate"   value="$(arg sample_rate)"/>
    <param name="swerve_id"     value="$(arg swerve_id)"/>
    <param name="wheel_id"      value="$(arg wheel_id)"/>
    <param name="wheel_reverse" value="$(arg wheel_reverse)"/>
    <param name="sim"           value="$(arg sim)"/>
  </node>
  <include file="$(find wheel_controller)/launch/ros_controllers.launch"/>

</launch>
